packages/plugins/graphql/server/services/utils/mappers/graphql-filters-to-strapi-query.js

```diff
@@ -68,13 +68,19 @@ module.exports = ({ strapi }) => {
 
       for (const [key, value] of Object.entries(filters)) {
         // If the key is an attribute, update the value
-        if (isAttribute(key)) {
+        if (isAttribute(key) || isComponent(attributes[key])) {
           const attribute = attributes[key];
-
           // If it's a scalar attribute
           if (virtualScalarAttributes.includes(key) || isStrapiScalar(attribute)) {
             // Replace (recursively) every GraphQL scalar operator with the associated Strapi operator
             resultMap[key] = recursivelyReplaceScalarOperators(value);
+          } else if (isComponent(attribute)) {
+            const replacedScalarOperators = recursivelyReplaceScalarOperators(value);
```

Comment:
If we want to handle deep filtering on more than two levels, we'll need to recursively replace every attribute of the component, not only the scalar ones.

What you could do is something really similar to how the relations are handled:

```js
// If it's a deep filter on a component
else if (isComponent(attribute)) {
  // Fetch the model from the component attribute
  const relModel = strapi.getModel(attribute.component);

  // Recursively apply the mapping to the value using the fetched model,
  // and update the value within `resultMap`
  resultMap[key] = this.graphQLFiltersToStrapiQuery(value, relModel);
}
```